Triangulation_3/demo/Triangulation_3/Viewer.cpp

```diff
@@ -1667,59 +1659,131 @@ void Viewer::drawFacet(const Triangle_3& t, std::vector<float> *vertices)
 
 void Viewer::drawWithNames()
 {
-    for(int i=0; i<m_pScene->m_vhArray.size(); ++i) {
-        // push a name for each point onto the name stack
-        // note: it can NOT be used between glBegin and glEnd
-        glPushName( i );
-
-        // draw the point
-        glBegin(GL_POINTS);
-        Point_3& p = m_pScene->m_vhArray.at(i)->point();
-        glVertex3f(p.x(), p.y(), p.z());
-        glEnd();
-
-        // pop one name off the top of the name stack
-        glPopName();
-    }//end-for-points
+  for(int i=0; i<m_pScene->m_vhArray.size(); ++i) {
+    //clear depth
+    glClearDepthf(1.0f);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    //draw point
+    Point_3& p = m_pScene->m_vhArray.at(i)->point();
+    GLfloat buf[3];
+    buf[0]=p.x();
+    buf[1]=p.y();
+    buf[2]=p.z();
+    rendering_program.bind();
+    vao[27].bind();
+    buffers[33].bind();
+    buffers[33].allocate(buf, 3*sizeof(GLfloat));
+    rendering_program.enableAttributeArray("vertex");
+    rendering_program.setAttributeArray("vertex",GL_FLOAT,0,3);
+    buffers[33].release();
+    vao[3].release();
+    
+    QMatrix4x4 mvpMatrix;
+    double mat[16];
+    camera()->getModelViewProjectionMatrix(mat);
+    for(int i=0; i < 16; i++)
+    {
+      mvpMatrix.data()[i] = (float)mat[i];
+    }
```

Comment:
There is a [warning], about that loop:
```
...\test\Triangulation_3_Demo\Viewer.cpp(1701) : warning C4258: 'i' : definition from the for loop is ignored; the definition from the enclosing scope is used
        ...\test\Triangulation_3_Demo\Viewer.cpp(1684) : definition of 'i' ignored
        ...\test\Triangulation_3_Demo\Viewer.cpp(1662) : definition of 'i' used
```
[warning]: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.13-Ic-62/Triangulation_3_Demo/TestReport_afabri_x64_Cygwin-Windows10_MSVC2013-Release-64bits.gz

Just rename the `i` of the second loop to something else, like `i2`, to avoid the warning.